repos/system_upgrade/el7toel8/actors/addupgradebootentry/libraries/library.py

```diff
@@ -2,22 +2,49 @@
 import re
 
 from leapp.exceptions import StopActorExecutionError
-from leapp.libraries.stdlib import api, run
+from leapp.libraries.common.config import architecture
+from leapp.libraries.stdlib import api, run, CalledProcessError
 from leapp.models import BootContent
 
 
 def add_boot_entry():
     debug = 'debug' if os.getenv('LEAPP_DEBUG', '0') == '1' else ''
 
     kernel_dst_path, initram_dst_path = get_boot_file_paths()
+    try:
+        _remove_old_upgrade_boot_entry(kernel_dst_path)
```

Comment:
won't grubby fail if kernel_dst_path doesn't exist?

Reply:
It's possible, but in such case we have completely different problem because this should not happen - when the actor is processed, the path should exists, otherwise there is an error in actor that prepares that is supposed to prepare it or stop the upgrade. That's same problem as in case we create the new entry in the next command.

In my testing, this currently works for the first run, even for another one.

Reply:
I meant that there is no bootloader entry yet for that kernel, not that the file does not exist.

IOW: You are trying to remove something that isn't there yet.

Reply:
And that's what is not doing any problem:
> In my testing, this currently works for the first run, even for another one.

So the boot entry is not there yet when I run that for the first time on completely new machine. It is question whether it is ok, but you can put any value (even nonexisting file) as the argument for `--remove-kernel` and it ends with exit code: `0`.